### DISCUSSION TITLE
Hot Reloading in Development

### DISCUSSION THREAD

**Discussion by michealroberts** [DavidVujic/python-polylith#190] [Q&A] [answered]
Absolutely throwing this one out to discussion, but one issue I find is that in development we have a separate serving mechanism from the project, e.g., we would run:

```
hypercorn example.api.core:app --bind ':5000' --reload --workers 1 
```

From inside the bases/example/api folder.

Whereas ideally, in local development we should be using the Dockerised configuration from the projects folder, but this currently lacks hot reloading as we need to run `poetry build-project` ... I'm wondering what QoL can be done here ...

**Comment by DavidVujic** [comment]
Probably a lot slower than the built-in hot reload feature of hypercorn, but how about something like `nodemon` (or a Python equivalent) to listen for file changes, and trigger a "build and restart docker" script?

**Comment by DavidVujic** [comment, marked as answer]
Another option could be to have a "local dev" Dockerfile (naming it something like `Dockerfile.dev`) that mounts a volume at your local host machine instead of the production Dockerfile that copies a built wheel. I think you can do that, and have it run with a reload option. Then you can write code, and the running app in docker would reload. My recommendation would still be to run the app on your local machine directly, for a better developer experience. 

(This thing is about Python development in general, and not specific to Polylith)
